tasks/release.py

```diff
@@ -533,17 +559,131 @@ def _get_highest_version_from_release_json(release_json, highest_major, version_
+            raise Exit("Aborting release.json update.", 1)
+
+    print(TAG_FOUND_TEMPLATE.format(repo_name, _stringify_version(version)))
+    return version
+
+
+def _confirm_independent_dependency_repo_version(repo, latest_version, highest_release_json_version):
+    """
+    Checks if the two versions of a repository we found (from release.json and from the available repo tags)
+    are different. If they are, asks the user for confirmation before updating the version.
+    """
+
+    confirmed_version = highest_release_json_version
+    if latest_version != highest_release_json_version:
+        print(
+            color_message(
+                DIFFERENT_TAGS_TEMPLATE.format(
+                    repo, _stringify_version(latest_version), _stringify_version(highest_release_json_version)
+                ),
+                "orange",
+            )
+        )
+        if yes_no_question(
+            "Do you want to update {} to {}?".format(repo, _stringify_version(latest_version)), "orange", False
+        ):
+            confirmed_version = latest_version
+
+    return confirmed_version
+
+
+def _fetch_independent_dependency_repo_version(
+    repo_name, release_json, agent_major_version, github_token, release_json_key
 ):
+    """
+    Fetches the latest tag on a given repository whose version scheme doesn't match the one used for the Agent:
+    - first, we get the latest version used in release entries of the matching Agent major version
+    - then, we fetch the latest version available in the repository
+    - if the above two versions are different, emit a warning and ask for user confirmation before updating the version.
+    """
+
+    highest_version = _get_highest_version_from_release_json(
+        release_json, agent_major_version, VERSION_RE, release_json_key=release_json_key,
+    )
+    # NOTE: This assumes that the repository doesn't change the way it prefixes versions.
+    version = _get_highest_repo_version(github_token, repo_name, highest_version["prefix"], None, VERSION_RE)
+    if not version:
+        raise Exit("Couldn't find any {} version.".format(repo_name), 1)
+
+    version = _confirm_independent_dependency_repo_version(repo_name, version, highest_version)
+    print(TAG_FOUND_TEMPLATE.format(repo_name, _stringify_version(version)))
 
-    highest_release_json_version = _get_highest_version_from_release_json(release_json, highest_major, version_re)
+    return version
+
+
+def _get_windows_ddnpm_release_json_info(
+    release_json, agent_major_version, version_re, is_first_rc=False,
+):
+    """
+    Gets the Windows NPM driver info from the previous entries in the release.json file.
+    """
 
     # First RC should use the data from nightly section otherwise reuse the last RC info
     if is_first_rc:
         print("Using 'nightly' DDNPM values")
         release_json_version_data = release_json['nightly']
     else:
+        highest_release_json_version = _get_highest_version_from_release_json(
```

Comment:
As with the previous comment, maybe `_get_highest_version_from_release_json` should raise an exception if it can't get the version. For this one we don't check everywhere (e.g. we don't in `_fetch_independent_dependency_repo_version`) but it feels like it won't harm to check every time